third_party/terraform/resources/resource_google_service_account.go

```diff
@@ -87,18 +89,35 @@ func resourceGoogleServiceAccountCreate(d *schema.ResourceData, meta interface{}
 
 	d.SetId(sa.Name)
 
-	err = retryTimeDuration(func() (operr error) {
-		_, saerr := config.clientIAM.Projects.ServiceAccounts.Get(d.Id()).Do()
-		return saerr
-	}, d.Timeout(schema.TimeoutCreate), isNotFoundRetryableError("service account creation"))
-
+	err = PollingWaitTime(resourceServiceAccountPollRead(d, meta), PollCheckForExistence, "Creating Service Account", d.Timeout(schema.TimeoutCreate))
 	if err != nil {
-		return fmt.Errorf("Error reading service account after creation: %s", err)
+		log.Printf("[ERROR] Unable to confirm eventually consistent Service Account %q finished updating: %q", d.Id(), err)
 	}
 
 	return resourceGoogleServiceAccountRead(d, meta)
 }
 
+func resourceServiceAccountPollRead(d *schema.ResourceData, meta interface{}) PollReadFunc {
+	return func() (map[string]interface{}, error) {
+		config := meta.(*Config)
+
+		url, err := replaceVars(d, config, "{{IAMDefaultBasePath}}projects/{{project}}/serviceAccounts/{{account_id}}")
+		if err != nil {
+			return nil, err
+		}
+
+		project, err := getProject(d, config)
+		if err != nil {
+			return nil, err
+		}
+		res, err := sendRequest(config, "GET", project, url, nil, pubsubTopicProjectNotReady)
```

Comment:
I think we need a different error predicate here. isNotFoundRetryableError should be good to guard against 404's